### PR TITLE
Port grouping redo/v26

### PR DESCRIPTION
**Pull request by inashivb**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) tickets:
- https://redmine.openinfosecfoundation.org/issues/6795
- https://redmine.openinfosecfoundation.org/issues/6792
- https://redmine.openinfosecfoundation.org/issues/6414

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1678

Previous PR: https://github.com/OISF/suricata/pull/10566

Changes since v25:
- add ticket 6414 to relevant commits

### CODE REVIEW

File: src/detect-engine-siggroup.c [hunk 1]
```diff
@@ -92,9 +92,12 @@ static SigGroupHeadInitData *SigGroupHeadInitDataAlloc(uint32_t size)
...
 
     /* initialize the signature bitarray */
-    sghid->sig_size = size;
-    if ((sghid->sig_array = SCCalloc(1, sghid->sig_size)) == NULL)
+    size = sghid->sig_size = size + 16 - (size % 16);
+    void *ptr = SCMallocAligned(sghid->sig_size, 16);
+    if (ptr == NULL)
         goto error;
+    memset(ptr, 0, size);
+    sghid->sig_array = ptr;
```
Comment (inashivb) [inline src/detect-engine-siggroup.c:100]: Sorry for being late.
Q: Why don't we only align the bitarray if SSE2 instruction set is available? `sig_size` would be different for example in such a case and we'll malloc lesser bytes.
e.g. for a size 10,
`master`: sig_size = 10; sizeof(sig_array) = 10 * sizeof(u8)
after this PR: sig_size = 16; sizeof(sig_array) = 16 * sizeof(u8) even when SSE2 is not available

Comment (inashivb) [reply src/detect-engine-siggroup.c:100]: ok. that works then

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 18954

**Comment by victorjulien** [comment]
Merged in #10569, thanks!

**Pull request closed without merging**